### PR TITLE
TF-523: Share a file from my device

### PR DESCRIPTION
**Pull request by ManhNTX**
Test Android:

https://user-images.githubusercontent.com/107173849/193953115-cbd81cf1-99d3-4918-93eb-cfd1907d4220.mov

Test IOS:

https://user-images.githubusercontent.com/107173849/194011667-eacd31a5-8b5d-4d7d-b500-1965e9ebd81d.mov

Test Inline:

https://user-images.githubusercontent.com/107173849/194214636-a6d63250-1550-418d-af6f-f5585fc4aba2.mov

### CODE REVIEW

**Comment by ManhNTX** [comment]
@hoangdat Can you test this PR on a real device (ios/android) ? Thank!!

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -90,6 +90,29 @@
                 <data android:mimeType="image/*" />
             </intent-filter>
 
+            <intent-filter>
+                <action android:name="android.intent.action.SEND" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="video/*" />
```
Comment (dab246) [inline android/app/src/main/AndroidManifest.xml:96]: IMO, This is not necessary. If you added `mimeType="*/*" />` it already supports all file types.

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -90,6 +90,29 @@
...
+            </intent-filter>
+
+            <intent-filter>
+                <action android:name="android.intent.action.SEND_MULTIPLE" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <data android:mimeType="video/*" />
```
Comment (dab246) [inline android/app/src/main/AndroidManifest.xml:102]: same

**Review by dab246 — COMMENTED** [review]
Please provide demo when app is killed

**Review by hoangdat — APPROVED** [review]

**Comment by hoangdat** [comment]
can not test in real iOS device, please merge and test in beta test

File: lib/main/utils/email_receive_manager.dart [hunk 3]
```diff
@@ -5,6 +5,9 @@ import 'package:rxdart/rxdart.dart';
 
 class EmailReceiveManager {
 
+  /// Regex check Url is mail or not
+  final mailRegex = RegExp(r'(([a-zA-Z0-9\-\_\.])+@[a-zA-Z\_]+?(\.[a-zA-Z]{2,6})+)');
```
Comment (dab246) [inline lib/main/utils/email_receive_manager.dart:9]: Why don't you use the function `GetUtils.isEmail()` of `Getx` to check if it is `email`?

**Pull request merged**